box-content-sdk/src/main/java/com/box/androidsdk/content/requests/BoxRequestsFile.java

```diff
@@ -1206,4 +1221,379 @@ public void setRequestedPage(int page) {
+                mFileInputStream.read(partBuffer);
+                mdPart.reset();
+                mdPart.update(partBuffer);
+                partSha1s.add(Base64.encodeToString(mdPart.digest(), Base64.DEFAULT));
+                mdFile.update(partBuffer);
+            }
+            uploadSession.setPartsSha1(partSha1s);
+            uploadSession.setSha1(Base64.encodeToString(mdFile.digest(), Base64.DEFAULT));
+        }
+
+        /**
+         * Set file name of the file to upload
+         * @param mFileName
+         */
+        public void setFileName(String mFileName) {
+            this.mFileName = mFileName;
+            mBodyMap.put("file_name", mFileName);
+        }
+
+        /**
+         * Returns the name of the file to upload.
+         *
+         * @return  name of the file to upload.
+         */
+        public String getFileName() {
+            return mFileName;
+        }
+
+        /**
+         * Returns the size of the file to upload
+         * @return file size in bytes
+         */
+        public long getFileSize() {
+            return mFileSize;
+        }
+
+        /**
+         * Returns the destination folder id for the uploaded file.
+         *
+         * @return  id of the destination folder for the uploaded file.
+         */
+        public String getDestinationFolderId() {
+            return mDestinationFolderId;
+        }
+
+    }
+
+    /**
+     * Request for uploading a session part
+     */
+    public static class UploadSessionPart extends BoxRequest<BoxUploadSessionPart, UploadSessionPart> {
+        private static final long serialVersionUID = 8245675031279971502L;
+        static final String DIGEST_HEADER = "digest";
+        private static final String CONTENT_RANGE_HEADER = "content-range";
+        static final String DIGEST_HEADER_PREFIX_SHA = "sha=";
+
+        final int mPartNumber;
+        final long mFileSize;
+        byte[] mData;
+        int mCurrentChunkSize;
+        final BoxUploadSession mUploadSession;
+
+        /**
+         *
+         * @param file Creates an upload file session part  with the default parameters
+         * @param uploadSession
+         * @param partNumber
+         * @param session
+         */
+        public UploadSessionPart(File file, BoxUploadSession uploadSession, int partNumber, BoxSession session) throws IOException {
+            super(BoxUploadSessionPart.class, uploadSession.getEndpoints().getUploadPartEndpoint(), session);
+            mRequestUrlString = uploadSession.getEndpoints().getUploadPartEndpoint();
+            mRequestMethod = Methods.PUT;
+            mPartNumber = partNumber;
+            mUploadSession = uploadSession;
+            mFileSize = file.length();
+            mCurrentChunkSize = BoxUploadSession.getChunkSize(uploadSession, partNumber, mFileSize);
+            //Read the partSize bytes from the stream
+            mData = new byte[mCurrentChunkSize];
```

Comment:
This mData byte array creates a big memory footprint for our app. (loading 8mb or more in RAM).
Instead please use the BoundedInputStream from Apache, and set the max to read as the offset + chunkSize to read.